Sources/src/FromCorsikaBinary_source.cc

```diff
@@ -137,13 +137,32 @@ namespace mu2e {
 
     //----------------------------------------------------------------
     unsigned CorsikaBinaryDetail::getSubRunNumber(const std::string& filename) const {
-      const std::string::size_type islash = filename.find("DAT") ;
-      const std::string basename = (islash == std::string::npos) ? filename : filename.substr(islash + 3);
+      const std::string::size_type corsikaConvention = filename.find("DAT");
```

Comment:
Hi Roberto,  I think the current patch will break on an input filename like
/mu2e/data/users/gandr/MYDATA/DAT1001
I suggest to use regular expressions to require a more precise match.
Something like:

#include <regex>

then in the getSubRun(filename) method:

      std::regex re_corsika("^(.*/)?DAT([0-9]+)$");
      std::regex re_mu2e("^(.*/)?sim\\.\\w+\\.[\\w-]+\\.[\\w-]+\\.([0-9]+)\\.csk$");

      std::string strrun;
      std::smatch match;
      if(std::regex_search(filename, match, re_corsika)) {
        // [0]: the whole string                                                                                                                                                                      
        // [1]: dirname or emtpy                                                                                                                                                                      
        // [2]: the run number string                                                                                                                                                                 
        sr = std::stoi(match.str(2));
      }   
      else if(std::regex_search(filename, match, re_mu2e)) {
        // [0]: the whole string                                                                                                                                                                      
        // [1]: dirname or emtpy                                                                                                                                                                      
        // [2]: the run number string                                                                                                                                                                 
        sr = std::stoi(match.str(2));
      }
      else {
         throw cet::exception("BADINPUT", " FromCorsikaBinary: ")
          << " Can not parse filename to extract subrun number:  "<<filename<<"\n";

      }

Reply:
You're right, one solution however would be to change `filename.find("DAT")` into `filename.find_last_of("DAT")`. Personally I prefer to use regexs only when they are absolutely necessary, because I find them unreadable (you basically have to take a leap of faith). I think in this case we could achieve the same result without using them, what do you think?